src/H5Dpkg.h

```diff
@@ -115,6 +115,13 @@
 /* Package Private Typedefs */
 /****************************/
 
+/* Typedef for cached dataset creation property list information */
+typedef struct H5D_dcpl_cache_t {
```

Comment:
This is not used outside of the H5D package so I demoted it to this header to minimize issues with circular dependencies